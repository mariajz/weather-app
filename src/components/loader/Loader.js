import React from 'react';
import { Animated } from 'react-native';
import { LoaderWrapper } from './Loader.style';
import { CircularLoader } from '../../../assets/animated-icons';

const Loader = () => {
    const animatedValue = new Animated.Value(0);

    Animated.loop(
        Animated.timing(animatedValue, {
            toValue: 1,
            duration: 1000,
            useNativeDriver: true,
        }),
    ).start();

    const interpolateValue = animatedValue.interpolate({
        inputRange: [0, 1],
        outputRange: ['0deg', '360deg'],
    });

    return (
        <LoaderWrapper>
            <Animated.View
                style={{ transform: [{ rotate: interpolateValue }] }}>
                <CircularLoader />
            </Animated.View>
        </LoaderWrapper>
    );
};

export default Loader;
