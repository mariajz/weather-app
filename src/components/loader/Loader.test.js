import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import Loader from './Loader';
import { Animated } from 'react-native';

jest.mock('../../../assets/animated-icons', () => ({
    CircularLoader: () => <MockView />,
}));

describe('Loader', () => {
    beforeEach(() => {
        const mockStart = jest.fn();
        jest.spyOn(Animated, 'loop').mockImplementation(() => ({
            start: mockStart,
        }));
    });

    it('should render Loader', () => {
        const container = render(<Loader />);

        expect(container).toMatchSnapshot();
    });
});
