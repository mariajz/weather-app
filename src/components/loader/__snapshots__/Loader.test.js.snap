// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render Loader 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      }
    }
  >
    <View />
  </View>
</View>
`;
